ViewModel/Study/FDACache.cs

```diff
@@ -544,6 +558,24 @@ public void UpdateIASElement(IASElementSet oldElement, IASElementSet newElement)
                 IASElementUpdated?.Invoke(this, new Saving.ElementUpdatedEventArgs(oldElement, newElement));
             }
         }
+        public void UpdateAlternativeElement(AlternativeElement oldElement, AlternativeElement newElement)
+        {
+            int index = -1;
+            for (int i = 0; i < AlternativeElements.Count; i++)
```

Comment:
Can you do this using IndexOf or some equivalent?  if oldElement is the same instance in your AlternativeElements then you should be able to find the index in that list.